pkg/engine/openapi.go

```diff
@@ -70,12 +83,17 @@ func (e *Engine) runOpenAPI(tool types.Tool, input string) (*Return, error) {
 		return nil, fmt.Errorf("failed to create request: %w", err)
 	}
 
-	// Check for a bearer token (only if using HTTPS)
-	if u.Scheme == "https" {
-		// For "https://example.com" the bearer token env name would be GPTSCRIPT_EXAMPLE_COM_BEARER_TOKEN
-		bearerEnv := "GPTSCRIPT_" + env.ToEnvLike(u.Hostname()) + "_BEARER_TOKEN"
-		if bearerToken, ok := envMap[bearerEnv]; ok {
-			req.Header.Set("Authorization", "Bearer "+bearerToken)
+	// Check for authentication (only if using HTTPS)
+	if u.Scheme == "https" && len(instructions.SecurityInfos) > 0 {
+		if err := handleAuths(req, envMap, instructions.SecurityInfos); err != nil {
+			return nil, fmt.Errorf("error setting up authentication: %w", err)
+		}
+
+		// If there is a bearer token set for the whole server, and no Authorization header has been defined, use it.
+		if token, ok := envMap["GPTSCRIPT_"+env.ToEnvLike(u.Hostname())+"_BEARER_TOKEN"]; ok {
```

Comment:
@g-linville hi, I just tried the openapi integration with this new code, and using a global _BEARER_TOKEN env var doesn't work anymore, because this code is not reached if I have a security info set, but without the specific _${name} env var.

for example in my case, my openapi document contains:

```yaml
components:
  securitySchemes:
    bearerAuth:
      bearerFormat: JWT
      scheme: bearer
      type: http
```

so if I don't have the `_BEARERAUTH` env var, I get an error - instead of the code trying to lookup the `_BEARER_TOKEN` env var.

that's not a big issue because I can change the env var on my side, but it's just that it might confuse people...

Reply:
@vbehar Thanks for bringing this to my attention. Yes, this was technically a breaking change, but there has been no release of GPTScript that includes the OpenAPI feature (the only way to use it has been to compile it yourself) so that's why I didn't specifically call it out anywhere.